pkg/secret/secret.go

```diff
@@ -0,0 +1,186 @@
+	"sort"
+	"time"
+)
+
+type (
+	Service interface {
+		ListNamesAndLabels() (resourceNames, labels []string, err error)
+		List(listOptions metav1.ListOptions) (resources []v1.Secret, err error)
+		GetUnused(namespace string, secrets []v1.Secret) (unusedSecrets []v1.Secret, funcErr error)
+		Delete(secrets []v1.Secret)
+		FilterByTime(secrets []v1.Secret, olderThan time.Time) (filteredSecrets []v1.Secret)
+		FilterByMaxCount(secrets []v1.Secret, keep int) (filteredSecrets []v1.Secret)
+		Print(secrets []v1.Secret)
+	}
+
+	SecretsService struct {
+		configuration Configuration
+		client        core.SecretInterface
+		helper        kubernetes.Kubernetes
+	}
+)
+
+type Configuration struct {
+	Batch bool
+}
+
+// NewSecretsService creates a new Service instance
+func NewSecretsService(client core.SecretInterface, helper kubernetes.Kubernetes, configuration Configuration) Service {
+	return SecretsService{
+		client:        client,
+		helper:        helper,
+		configuration: configuration,
+	}
+}
+
+// ListNamesAndLabels return the names and labels of all secrets
+func (ss SecretsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	secrets, err := ss.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, s := range secrets {
+		objectMetas = append(objectMetas, s.ObjectMeta)
+	}
+	secretNames, labels := util.GetNamesAndLabels(objectMetas)
+	return secretNames, labels, nil
+}
+
+// List returns a list of secrets from a namespace
+func (ss SecretsService) List(listOptions metav1.ListOptions) ([]v1.Secret, error) {
+	secrets, err := ss.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+	return secrets.Items, nil
+}
+
+// GetUnused returns unused resources
+func (ss SecretsService) GetUnused(namespace string, resources []v1.Secret) (unusedResources []v1.Secret, funcErr error) {
+	var usedSecrets []v1.Secret
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(resources, func(secret v1.Secret) {
+
+			secretName := secret.GetName()
+
+			if funk.Contains(usedSecrets, secret) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := ss.helper.ResourceContains(namespace, secretName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedSecrets = append(usedSecrets, secret)
+			}
+		})
+	})
+
+	for _, resource := range resources {
+		if !funk.Contains(usedSecrets, resource) {
+			unusedResources = append(unusedResources, resource)
+		}
+	}
+
+	return unusedResources, funcErr
+}
+
+// Delete removes secrets
+func (ss SecretsService) Delete(secrets []v1.Secret) {
+	for _, resource := range secrets {
+		namespace := resource.GetNamespace()
+		kind := "Secret"
+		name := resource.GetName()
+
+		if ss.configuration.Batch {
+			fmt.Println(resource.GetName())
+		} else {
+			log.Infof("Deleting %s %s/%s", kind, namespace, name)
```

Comment:
the `kind` variable is only used once. I would just write `log.Infof("Deleting Secret %s/%s", namespace, name)`

Reply:
fixed